com.hazelcast.jet.core.metrics.JobMetrics_StreamTest.when_jobRestarted_then_metricsReset_withJournal[0: mode=member]

_3.2_ (commit ad691bcbf6b0b9f0ba8a8ceda7b136b0ab91fca8)

Failed on Zulu JDK 8: http://jenkins.hazelcast.com/view/Jet%20-%20Release/job/release-jet-oss-master-zulu-jdk8/4/testReport/com.hazelcast.jet.core.metrics/JobMetrics_StreamTest/when_jobRestarted_then_metricsReset_withJournal_0__mode_member_/

Stacktrace:
```
java.lang.AssertionError: expected:<2> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.jet.core.metrics.JobMetrics_StreamTest.lambda$when_jobRestarted_then_metricsReset_withJournal$23(JobMetrics_StreamTest.java:191)
	at com.hazelcast.jet.core.JetTestSupport$1.run(JetTestSupport.java:203)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1310)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1412)
	at com.hazelcast.jet.core.JetTestSupport.assertTrueEventually(JetTestSupport.java:160)
	at com.hazelcast.jet.core.metrics.JobMetrics_StreamTest.when_jobRestarted_then_metricsReset_withJournal(JobMetrics_StreamTest.java:191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Hiccups measured while running test 'when_jobRestarted_then_metricsReset_withJournal[0: mode=member](com.hazelcast.jet.core.metrics.JobMetrics_StreamTest):'
20:01:55, accumulated pauses: 793 ms, max pause: 107 ms, pauses over 1000 ms: 0
20:02:00, accumulated pauses: 573 ms, max pause: 38 ms, pauses over 1000 ms: 0
20:02:05, accumulated pauses: 422 ms, max pause: 92 ms, pauses over 1000 ms: 0
20:02:10, accumulated pauses: 359 ms, max pause: 53 ms, pauses over 1000 ms: 0
20:02:15, accumulated pauses: 346 ms, max pause: 48 ms, pauses over 1000 ms: 0
20:02:20, accumulated pauses: 228 ms, max pause: 9 ms, pauses over 1000 ms: 0
20:02:25, accumulated pauses: 359 ms, max pause: 21 ms, pauses over 1000 ms: 0
20:02:30, accumulated pauses: 235 ms, max pause: 31 ms, pauses over 1000 ms: 0
20:02:35, accumulated pauses: 344 ms, max pause: 53 ms, pauses over 1000 ms: 0
20:02:40, accumulated pauses: 314 ms, max pause: 37 ms, pauses over 1000 ms: 0
20:02:45, accumulated pauses: 291 ms, max pause: 11 ms, pauses over 1000 ms: 0
20:02:50, accumulated pauses: 216 ms, max pause: 37 ms, pauses over 1000 ms: 0
20:02:55, accumulated pauses: 153 ms, max pause: 7 ms, pauses over 1000 ms: 0
20:03:00, accumulated pauses: 218 ms, max pause: 35 ms, pauses over 1000 ms: 0
20:03:05, accumulated pauses: 230 ms, max pause: 63 ms, pauses over 1000 ms: 0
20:03:10, accumulated pauses: 213 ms, max pause: 7 ms, pauses over 1000 ms: 0
20:03:15, accumulated pauses: 304 ms, max pause: 16 ms, pauses over 1000 ms: 0
20:03:20, accumulated pauses: 440 ms, max pause: 12 ms, pauses over 1000 ms: 0
20:03:25, accumulated pauses: 180 ms, max pause: 15 ms, pauses over 1000 ms: 0
20:03:30, accumulated pauses: 287 ms, max pause: 21 ms, pauses over 1000 ms: 0
20:03:35, accumulated pauses: 249 ms, max pause: 15 ms, pauses over 1000 ms: 0
20:03:40, accumulated pauses: 180 ms, max pause: 9 ms, pauses over 1000 ms: 0
20:03:45, accumulated pauses: 217 ms, max pause: 30 ms, pauses over 1000 ms: 0
20:03:50, accumulated pauses: 175 ms, max pause: 10 ms, pauses over 1000 ms: 0
20:03:55, accumulated pauses: 268 ms, max pause: 37 ms, pauses over 1000 ms: 0
20:04:00, accumulated pauses: 244 ms, max pause: 141 ms, pauses over 1000 ms: 0
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_StreamTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_StreamTest.java
@@ -62,9 +62,8 @@ public class JobMetrics_StreamTest extends TestInClusterSupport {
         putIntoMap(map, 2, 1);
         List<String> sink = jet().getList(sinkListName);
 
-        Pipeline p = createPipeline();
         // When
-        Job job = jet().newJob(p);
+        Job job = jet().newJob(createPipeline());
 
         assertTrueEventually(() -> assertEquals(2, sink.size()));
         // Then
@@ -82,8 +81,7 @@ public class JobMetrics_StreamTest extends TestInClusterSupport {
         putIntoMap(map, 2, 1);
         List<String> sink = jet().getList(sinkListName);
 
-        Pipeline p = createPipeline();
-        Job job = jet().newJob(p, JOB_CONFIG_WITH_METRICS);
+        Job job = jet().newJob(createPipeline(), JOB_CONFIG_WITH_METRICS);
 
         putIntoMap(map, 1, 1);
 
@@ -102,13 +100,11 @@ public class JobMetrics_StreamTest extends TestInClusterSupport {
         putIntoMap(map, 2, 1);
         List<String> sink = jet().getList(sinkListName);
 
-        Pipeline p = createPipeline();
-
         JobConfig jobConfig = new JobConfig()
             .setStoreMetricsAfterJobCompletion(true)
             .setProcessingGuarantee(ProcessingGuarantee.EXACTLY_ONCE);
         // When
-        Job job = jet().newJob(p, jobConfig);
+        Job job = jet().newJob(createPipeline(), jobConfig);
 
         putIntoMap(map, 1, 1);
 
@@ -146,8 +142,7 @@ public class JobMetrics_StreamTest extends TestInClusterSupport {
         putIntoMap(map, 2, 1);
         List<String> sink = jet().getList(sinkListName);
 
-        Pipeline p = createPipeline();
-        Job job = jet().newJob(p, JOB_CONFIG_WITH_METRICS);
+        Job job = jet().newJob(createPipeline(), JOB_CONFIG_WITH_METRICS);
 
         assertTrueEventually(() -> assertEquals(2, sink.size()));
         assertTrueEventually(() -> assertMetrics(job.getMetrics(), 3, 1));
@@ -177,11 +172,9 @@ public class JobMetrics_StreamTest extends TestInClusterSupport {
     @Test
     public void when_jobRestarted_then_metricsReset_withJournal() {
         Map<String, String> map = jet().getMap(journalMapName);
-        putIntoMap(map, 2, 1);
         List<String> sink = jet().getList(sinkListName);
 
-        Pipeline p = createPipeline(JournalInitialPosition.START_FROM_CURRENT);
-        Job job = jet().newJob(p, JOB_CONFIG_WITH_METRICS);
+        Job job = jet().newJob(createPipeline(), JOB_CONFIG_WITH_METRICS);
 
         assertJobStatusEventually(job, RUNNING);
         assertTrueEventually(() -> assertMetrics(job.getMetrics(), 0, 0));
@@ -195,28 +188,24 @@ public class JobMetrics_StreamTest extends TestInClusterSupport {
         job.restart();
 
         assertJobStatusEventually(job, RUNNING);
-        assertTrueEventually(() -> assertEquals(2, sink.size()));
+        assertTrueEventually(() -> assertEquals(4, sink.size()));
         // Then
-        assertTrueEventually(() -> assertMetrics(job.getMetrics(), 0, 0));
+        assertTrueEventually(() -> assertMetrics(job.getMetrics(), 3, 1));
 
         putIntoMap(map, 1, 1);
-        assertTrueEventually(() -> assertEquals(3, sink.size()));
-        assertTrueEventually(() -> assertMetrics(job.getMetrics(), 2, 1));
+        assertTrueEventually(() -> assertEquals(5, sink.size()));
+        assertTrueEventually(() -> assertMetrics(job.getMetrics(), 5, 2));
 
         job.cancel();
         assertJobStatusEventually(job, FAILED);
-        assertMetrics(job.getMetrics(), 2, 1);
+        assertMetrics(job.getMetrics(), 5, 2);
     }
 
     private Pipeline createPipeline() {
-        return createPipeline(JournalInitialPosition.START_FROM_OLDEST);
-    }
-
-    private Pipeline createPipeline(JournalInitialPosition position) {
         Pipeline p = Pipeline.create();
-        p.<Map.Entry<String, String>>readFrom(Sources.mapJournal(journalMapName, position))
+        p.<Map.Entry<String, String>>readFrom(Sources.mapJournal(journalMapName, JournalInitialPosition.START_FROM_OLDEST))
                 .withIngestionTimestamps()
-                .map(map -> map.getKey())
+                .map(Map.Entry::getKey)
                 .filter(word -> !word.startsWith(FILTER_OUT_PREFIX))
                 .writeTo(Sinks.list(sinkListName));
         return p;

